src/Parsers/ASTInsertQuery.cpp

```diff
@@ -110,30 +105,41 @@ void ASTInsertQuery::formatImpl(const FormatSettings & settings, FormatState & s
     ///     SETTINGS max_threads=1
     ///     VALUES
     ///
-    char delim = settings_ast ? settings.nl_or_ws : ' ';
-
     if (select)
     {
-        settings.ostr << delim;
-        select->formatImpl(settings, state, frame);
+        if (settings_ast)
+            out.nlOrWs();
+        else
+            out.ostr << ' ';
```

Comment:
Now it became more complicated. We can add a lambda function `put_delimiter()` and write here

```
put_delimiter();
select->formatImpl(out);
```